pkg/config/mock/mock.go

```diff
@@ -0,0 +1,71 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package mock
+
+import (
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/config/setup"
+)
+
+var (
+	isConfigMocked = false
+	m              = sync.Mutex{}
+)
+
+// MockConfig should only be used in tests
+type MockConfig struct {
+	model.Config
+}
+
+// Set is used for setting configuration in tests
+func (c *MockConfig) Set(key string, value interface{}, source model.Source) {
+	c.Config.Set(key, value, source)
+}
+
+// SetWithoutSource is used for setting configuration in tests
+func (c *MockConfig) SetWithoutSource(key string, value interface{}) {
+	c.Config.SetWithoutSource(key, value)
+}
+
+// SetKnown is used for setting configuration in tests
+func (c *MockConfig) SetKnown(key string) {
+	c.Config.SetKnown(key)
+}
+
+// Mock is creating and returning a mock config
+func New(t testing.TB) *MockConfig {
+	// We only check isConfigMocked when registering a cleanup function. 'isConfigMocked' avoids nested calls to
+	// Mock to reset the config to a blank state. This way we have only one mock per test and test helpers can call
+	// Mock.
```

Comment:
Do you have more context about this comment?
In unit tests, you mostly want to have a refresh instance each time you call `New`?

Reply:
This code is just moved from `pkg/config/mock.go` [here](https://github.com/DataDog/datadog-agent/pull/27482/files/0f3054ac10a29efd985a741467c0ddd15fc83255#diff-39d7ce767a00785ac9ed9c756fb0659948cbf220cbe33dc1e186dcaf6ea7ed01L44). This is used throughout the code base so test helpers can mock the config even if it was mocked before by the caller.